sys/net/application_layer/gcoap/gcoap.c

```diff
@@ -91,14 +91,37 @@ static void *_event_loop(void *arg)
 
         if (res > 0) {
             switch (msg_rcvd.type) {
-                case GCOAP_MSG_TYPE_TIMEOUT:
-                    _expire_request((gcoap_request_memo_t *)msg_rcvd.content.ptr);
-                    break;
-                case GCOAP_MSG_TYPE_INTR:
-                    /* next _listen() timeout will account for open requests */
-                    break;
-                default:
-                    break;
+            case GCOAP_MSG_TYPE_TIMEOUT: {
+                gcoap_request_memo_t *memo = (gcoap_request_memo_t *)msg_rcvd.content.ptr;
+
+                /* no retries remaining */
+                if (memo->send_limit == GCOAP_SEND_LIMIT_NON
+                        || memo->send_limit == 0) {
+                    _expire_request(memo);
+                }
+                /* retries remaining; double timeout and resend */
+                else {
+                    unsigned i       = COAP_MAX_RETRANSMIT - memo->send_limit + 1;
```

Comment:
if you put the decrement, i.e. `memo->send_limit--`, above this statement you can omit the `+ 1` at the end of this line.

Reply:
I know its likely micro-optimisation, and the compiler will do that already. However, IMHO it makes the code also more readable.

Reply:
Yes, it's fine and clear with the decrement first. I had it in my head to increase timeout first, but it doesn't matter.